src/react/hooks/paragon/useParagonThemeCore.test.js

```diff
@@ -0,0 +1,106 @@
+      themeCore: {
+        urls: { default: 'https://cdn.jsdelivr.net/npm/@edx/paragon@$paragonVersion/dist/core.min.css' },
+      },
+      onLoad: themeOnLoad,
+    };
+    renderHook(() => useParagonThemeCore(coreConfig));
+    const createdLinkTag = document.head.querySelector('link');
+    act(() => createdLinkTag.onload());
+    expect(createdLinkTag.href).toBe(coreConfig.themeCore.urls.default);
+    expect(themeOnLoad).toHaveBeenCalledTimes(1);
+    expect(initialState).toBeTruthy();
+  });
+
+  it('should load the core default and brand url and change the loading state to true', () => {
+    themeOnLoad.mockImplementation(() => { initialState = true; });
+    const coreConfig = {
+      themeCore: {
+        urls: {
+          default: 'https://cdn.jsdelivr.net/npm/@edx/paragon@$paragonVersion/dist/core.min.css',
+          brandOverride: 'https://cdn.jsdelivr.net/npm/@edx/brand@$brandVersionVersion/dist/core.min.css',
+        },
+      },
+      onLoad: themeOnLoad,
+    };
+    renderHook(() => useParagonThemeCore(coreConfig));
+    const createdLinkTag = document.head.querySelector('link[data-paragon-theme-core="true"]');
+    const createdBrandLinkTag = document.head.querySelector('link[data-brand-theme-core="true"]');
+
+    act(() => { createdLinkTag.onload(); createdBrandLinkTag.onload(); });
+    expect(createdLinkTag.href).toBe(coreConfig.themeCore.urls.default);
+    expect(createdBrandLinkTag.href).toBe(coreConfig.themeCore.urls.brandOverride);
+    expect(themeOnLoad).toHaveBeenCalledTimes(1);
+    expect(initialState).toBeTruthy();
+  });
+
+  it('should dispatch a log error and fallback to PARAGON_THEME if can not load the core theme link', () => {
+    global.PARAGON_THEME = {
+      paragon: {
+        version: '1.0.0',
+        themeUrls: {
+          core: {
+            fileName: 'core.min.css',
+          },
+          variants: {
+            light: {
+              fileName: 'light.min.css',
+              default: true,
+              dark: false,
+            },
+          },
+        },
+      },
+    };
+    themeOnLoad.mockImplementation(() => { initialState = true; });
+    const coreConfig = {
+      themeCore: {
+        urls: {
+          default: 'https://cdn.jsdelivr.net/npm/@edx/paragon@$paragonVersion/dist/core.min.css',
+        },
+      },
+      onLoad: themeOnLoad,
+    };
+    renderHook(() => useParagonThemeCore(coreConfig));
+    const createdLinkTag = document.head.querySelector('link[data-paragon-theme-core="true"]');
+
+    act(() => { createdLinkTag.onerror(); });
+    expect(logError).toHaveBeenCalledTimes(1);
+    expect(logError).toHaveBeenCalledWith(`Failed to load core theme CSS from ${coreConfig.themeCore.urls.default}`);
+    expect(document.querySelector('link').href).toBe(`${getConfig().BASE_URL}/${PARAGON_THEME.paragon.themeUrls.core.fileName}`);
+    expect(initialState).toBeFalsy();
+  });
+
+  it('should not create any core link if is properly configured', () => {
```

Comment:
[nit/clarification] What does "properly configured" mean here to make it not create the core link? Might be good to make the test case name a bit more explicit.

Reply:
maybe like `should not create any core link if can not find urls definition` understanding this in the code
```
   // If there is no config for the core theme url, do nothing.
    if (!themeCore?.urls) {
```